File: Queue/list.c

#include <stdio.h>
#include <stdbool.h>
#include <limits.h>

typedef struct node {
    int data;
    struct node *next;
}node;

typedef struct {
    node* head;
    node* tail;    
}queue;

void init_queue(queue *q){
    q->head = NULL;
    q->tail = NULL;
};

void enqueue(queue *q , int data){

    node *new_node = malloc(sizeof(node));
    new_node->data = data;
    new_node->next = NULL;

    if(!new_node){
        return;
    }
    
    if(q->tail != NULL){
        q->tail->next = new_node;
    }

    q->tail = new_node;

    if(q->head == NULL){
        q->head = new_node;
    }
};

void dequeue(queue *q){

    if(q->head == NULL){
        return;
    }

    node *temp = q->head;

    int res = temp->data;
    q->head = q->head->next;

    if(q->head == NULL){
        q->tail = NULL;
    }

    free(temp);

    printf("%d",res);
};

void isEmptyorFull(queue *q){

   if (q->head != NULL)
   {
       printf("Not empty");
   }       
   else
   {
       printf("Empty");
   }      
};

void printList(queue *q){
    while (q->head != NULL)
    {
        printf("%d",q->head->data);
        q->head = q->head->next;
    }    
};

void main()
{
    queue s1 ;

    init_queue(&s1);    

    enqueue(&s1,10);
    enqueue(&s1,11);
    enqueue(&s1,12);
    enqueue(&s1,13);
    enqueue(&s1,14);
    enqueue(&s1,15);

    isEmpty(&s1);

    printList(&s1);    
};

File: LinkedList/SinglyLinkedList/List.py

class Node:
	def __init__(self,data):
		self.data = data
		self.next = None

class List:
	def __init__(self):
		self.head = None

	def append(self,data):
		temp = self.head
		new_node = Node(data)

		if(temp == None):
			temp = new_node
			return

		while temp.next is not None:
			temp = temp.next

		temp.next = new_node

	def get_length(self):
		if(self.head==None):
			return
		else:
			counter=0

			while (self.head is not None):
				self.head=self.head.next
				counter+=1
			print(counter)

	def delete_at(self,data):
		last = self.head

		if(self.head == None):
			print("No list Created...")
			return	
		else:
			if(self.head.data == data):
				head = last.next
				del(head)
				return

			while ():
				pass



	def print(self):
		temp = self.head

		while (temp is not None):
			print(temp.data)
			temp = temp.next

list = List()
list.head = Node(0)

list.append(1)
list.append(2)
list.append(3)
list.append(4)

list.print()
list.get_length()